MoreLinq.Test/RandomTest.cs

```diff
@@ -17,74 +17,86 @@ public class RandomTest
+            {
+                const int maxValue = 10;
 
-            Assert.Greater( maxValue, 0 );
-            MoreEnumerable.Random(null, maxValue);
+                Assert.Greater(maxValue, 0);
+                MoreEnumerable.Random(null, maxValue);
+            });
         }
 
         /// <summary>
         /// Verify that passing an <c>null</c> random generator results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestNullGeneratorException3()
         {
-            const int minValue = 10;
-            const int maxValue = 100;
-
-            Assert.LessOrEqual(minValue, maxValue);
-            MoreEnumerable.Random(null, minValue, maxValue);
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                const int minValue = 10;
+                const int maxValue = 100;
+
+                Assert.LessOrEqual(minValue, maxValue);
+                MoreEnumerable.Random(null, minValue, maxValue);
+            });
         }
 
         /// <summary>
         /// Verify that passing an <c>null</c> random generator results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestNullGeneratorException4()
         {
-            MoreEnumerable.RandomDouble(null);
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                MoreEnumerable.RandomDouble(null);
+            });
         }
 
         /// <summary>
         /// Verify that passing a negative maximum value yields an exception
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         public void TestNegativeMaxValueException()
         {
-            const int maxValue = -10;
-            Assert.Less( maxValue, 0 );
-            MoreEnumerable.Random(maxValue);
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                const int maxValue = -10;
+                Assert.Less(maxValue, 0);
+                MoreEnumerable.Random(maxValue);
+            });
         }
 
         /// <summary>
         /// Verify that passing lower bound that is greater than the upper bound results
         /// in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         public void TestMinValueGreaterThanMaxValueException()
         {
-            const int minValue = 100;
-            const int maxValue = 10;
-
-            Assert.Greater(minValue, maxValue);
-            MoreEnumerable.Random(minValue, maxValue);
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                const int minValue = 100;
+                const int maxValue = 10;
+
+                Assert.Greater(minValue, maxValue);
+                MoreEnumerable.Random(minValue, maxValue);
```

Comment:
Only this line should be inside the lambda to Assert.Throws as that's what will cause and being tested for an exception.